### PR TITLE
Remove mentions of "Spacewalk" and "Satellite" from API docs

### PR DESCRIPTION
**Pull request by renner**
## What does this PR change?

This patch is removing mentions of "Spacewalk" and "Satellite" from the API docs. Those mentions are confusing to users as neither Spacewalk nor Satellite is involved. General algorithm that is followed here:

- API docs: "Satellite" (the product) and "Satellite admin" (the role) -> "Uyuni" and "Uyuni admin" (via the `#product()` macro).
- In corresponding Javadoc comments: avoid mentions of the product name and use a more generic wording, use `product admin` when referring to the role.

TODO:

- [x] Remove the deprecated `satellite` namespace (`SatelliteHandler.java`).
- [x] Make sure that the handler removal is added to the release notes for upcoming releases (ping @juliogonzalez).
- [x] Replace "Red Hat Errata" with "patches".
- [x] Take advantage of the existing #product() macro.
- [ ] Understand what is done in `OrgHandler.setOrgConfigManagedByOrgAdmin()` and why it has an impact on performance.
- [ ] Create a separate issue to actually replace `satellite_admin` in the code and database (maybe with `product_admin`?).

## GUI diff

No difference.

- [X] **DONE**

## Documentation

- No documentation needed, this patch is actually changing the API docs.

- [X] **DONE**

## Test coverage

- No tests needed, only API docs are affected.

- [X] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/12689.

- [X] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below).

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"
- [ ] Re-run test "java_pgsql_tests" 
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Review by Loquacity — APPROVED** [review]
LGTM

**Review by cbosdo — APPROVED** [review]
:+1:

**Comment by renner** [comment]
@mcalmer and @paususe: this should be ready for a review now, I was happy to learn that we already have a `#product()` macro for the API docs (thanks to @cbosdo) and changed the patch accordingly.

File: java/code/src/com/redhat/rhn/frontend/xmlrpc/org/OrgHandler.java [hunk 1]
```diff
@@ -314,7 +314,7 @@ private ChannelFamily lookupChannelFamily(String channelFamilyLabel) {
 
     /**
      * Migrate systems from one organization to another.  If executed by
-     * a Satellite administrator, the systems will be migrated from their current
+     * a product administrator, the systems will be migrated from their current
```
Comment (mcalmer) [inline java/code/src/com/redhat/rhn/frontend/xmlrpc/org/OrgHandler.java:317]:
```suggestion
     * a #product() administrator, the systems will be migrated from their current
```

But I am unsure if this macro works in normal Java

File: java/code/src/com/redhat/rhn/frontend/xmlrpc/org/OrgHandler.java [hunk 2]
```diff
@@ -327,15 +327,15 @@ private ChannelFamily lookupChannelFamily(String channelFamilyLabel) {
      * @return list of systems migrated.
      * @throws FaultException A FaultException is thrown if:
      *   - The user performing the request is not an organization administrator
-     *   - The user performing the request is not a satellite administrator, but the
+     *   - The user performing the request is not a product administrator, but the
```
Comment (mcalmer) [inline java/code/src/com/redhat/rhn/frontend/xmlrpc/org/OrgHandler.java:330]:
```suggestion
     *   - The user performing the request is not a #product() administrator, but the
```

But I am unsure if this macro works in normal Java

File: java/code/src/com/redhat/rhn/frontend/xmlrpc/user/external/UserExternalHandler.java [hunk 3]
```diff
@@ -60,7 +60,7 @@
      *
      * @xmlrpc.doc Set whether we should keeps roles assigned to users because of
      * their IPA groups even after they log in through a non-IPA method. Can only be
-     * called by a satellite_admin.
+     * called by a #product() admin.
```
Comment (mcalmer) [inline java/code/src/com/redhat/rhn/frontend/xmlrpc/user/external/UserExternalHandler.java:63]: `satellite_admin` is the role label in DB. The text version is `SUSE Manager Administrator`.
So maybe we write not "#product() admin" but "#product() Administrator" ?

Here and below.

**Review by mcalmer — CHANGES_REQUESTED** [review]
Some questions about the changes insite

File: java/code/src/com/redhat/rhn/frontend/xmlrpc/org/OrgHandler.java (hunk 1 above)
Comment (renner) [reply java/code/src/com/redhat/rhn/frontend/xmlrpc/org/OrgHandler.java:317]: > But I am unsure if this macro works in normal Java

I guess we could make it work, but currently - as far as I know - we are not actually building and publishing javadoc. And in general we would rather want to go away from using velocity, so I would probably not want to introduce this macro for being used in javadoc comments. Therefore I purposely chose to use just "product administrator" here, this way it should be valid in the context of Uyuni, but also downstream, while developers hopefully know what is meant (the current `SAT_ADMIN` aka `satellite_administrator` in the DB).

Additionally I had the idea to actually replace `satellite_admin` throughout the DB and codebase with `product_admin` (or maybe something else?) to be more generic. If we did this, that comment would not need to be updated. I would be curious to know your opinion about that?

File: java/code/src/com/redhat/rhn/frontend/xmlrpc/org/OrgHandler.java (hunk 2 above)
Comment (renner) [reply java/code/src/com/redhat/rhn/frontend/xmlrpc/org/OrgHandler.java:330]: I did not use the macro on purpose, see my previous comment for the explanation.

File: java/code/src/com/redhat/rhn/frontend/xmlrpc/user/external/UserExternalHandler.java (hunk 3 above)
Comment (renner) [reply java/code/src/com/redhat/rhn/frontend/xmlrpc/user/external/UserExternalHandler.java]: Thanks, I added another commit (6d4baa4) with those changes, turning also additional occurrences of `Sat admin` to `product admin` in javadoc.

File: java/code/src/com/redhat/rhn/frontend/xmlrpc/org/OrgHandler.java (hunk 1 above)
Comment (mcalmer) [reply java/code/src/com/redhat/rhn/frontend/xmlrpc/org/OrgHandler.java:317]: As nobody read this I am fine with keeping it.

**Review by mcalmer — APPROVED** [review]

**Pull request merged**